clustering/ambari_cluster_state.py

```diff
@@ -0,0 +1,387 @@
+      The name of the cluster in web server
+    required: yes
+  cluster_name:
+    description:
+      The name of the cluster in ambari
+    required: yes
+  cluster_state:
+    description:
+      The desired state for the ambari cluster ['present', 'absent', 'started', 'stopped']. Setting the cluster
+      state to absent will first stop the cluster.
+    required: yes
+  blueprint_yml:
+    description:
+      The path to the file defining the cluster blueprint and host mapping. Required when state == 'present'
+    required: no
+  blueprint_name:
+    description:
+      The name of the blueprint. Required when state == 'present'
+    required: no
+  wait_for_complete:
+    description:
+      Whether to wait for the request to complete before returning. Default is False.
+    required: no
+'''
+
+EXAMPLES = '''
+# must use full relative path to any files in stored in roles/role_name/files/
+- name: Create a new ambari cluster
+    ambari_cluster_state:
+      host: localhost
+      port: 8080
+      username: admin
+      password: admin
+      cluster_name: my_cluster
+      cluster_state: present
+      blueprint_yml: roles/my_role/files/blueprint.yml
+      blueprint_name: hadoop
+      wait_for_complete: True
+- name: Start the ambari cluster
+  ambari_cluster_state:
+    host: localhost
+    port: 8080
+    username: admin
+    password: admin
+    cluster_name: my_cluster
+    cluster_state: started
+    wait_for_complete: True
+- name: Stop the ambari cluster
+  ambari_cluster_state:
+    host: localhost
+    port: 8080
+    username: admin
+    password: admin
+    cluster_name: my_cluster
+    cluster_state: stopped
+    wait_for_complete: True
+- name: Delete the ambari cluster
+  ambari_cluster_state:
+    host: localhost
+    port: 8080
+    username: admin
+    password: admin
+    cluster_name: my_cluster
+    cluster_state: absent
+'''
+__author__ = 'mbittmann'
+from ansible.module_utils.basic import *
+import json
+import os
+try:
+    import requests
+except ImportError:
+    REQUESTS_FOUND = False
+else:
+    REQUESTS_FOUND = True
+
+try:
+    import yaml
+except ImportError:
+    YAML_FOUND = False
+else:
+    YAML_FOUND = True
+
+def main():
+
+    argument_spec = dict(
+        host=dict(type='str', default=None, required=True),
+        port=dict(type='int', default=None, required=True),
+        username=dict(type='str', default=None, required=True),
+        password=dict(type='str', default=None, required=True),
+        cluster_name=dict(type='str', default=None, required=True),
+        cluster_state=dict(type='str', default=None, required=True,
+                           choices=['present', 'absent', 'started', 'stopped']),
+        blueprint_var=dict(type='dict', required=False),
+        blueprint_name=dict(type='str', default=None, required=False),
+        wait_for_complete=dict(default=False, required=False, choices=BOOLEANS),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec
```

Comment:
One question on this suggestion. The variables `blueprint_var` and `blueprint_name` are both required only when `cluster_state=='present'`. I'm not sure if I can implement the condition using the `required_together` param. My solution would be to use `required_together=['blueprint_name', 'blueprint_var']` since they are only every used together. This would not enforce the requirement along with cluster_state however.